docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+
+For a Swap Lit Action, both parties need to deposit their funds to the PKP and based on the conditions the PKP should be completing the swap:
+
+- Alice (or Bob) creates a Mint/Grant/Burn PKP
+- Alice deposits 4 token A on Base Sepolia to the PKP
+- Bob deposits 8 token B on Chronicles to the PKP
+- Alice (or Bob) executes the Lit Action.
+
+Lit Action execution, checks conditions and executes the swap from the PKP
+
+If the swap conditions aren't met then the Lit Action execution shall revert the funds to their depositors.
+
+### Architecture for Mint/Grant/Burn
+
+![accessControl](/img/mgb-flow.png)
+
+## 2) Lit Action
+
+Now we'll dive into the depths of the Lit Action and explore how we can conduct conditional signing.
+
+Below are the parameters that decide the swapping conditions.
+
+```js
+const chainAParams = {
+    from: "0x48e6a467852Fa29710AaaCDB275F85db4Fa420eB",
+    to: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    tokenAddress: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    chain: "baseSepolia",
+    amount: "4",
+    decimals: 18,
+    chainId: 84532,
+};
+
+const chainBParams = {
+    from: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    to: "0x48e6a467852Fa29710AaaCDB275F85db4Fa420eB",
+    tokenAddress: "0x2dcA1a80c89c81C37efa7b401e2d20c1ED99C72F",
+    chain: "yellowstone",
+    amount: "8",
+    decimals: 18,
+    chainId: 175188,
+};
+/* 
+deposit1: wallet A deposits on chain B, if action executes, funds are transferred to wallet B
+deposit2: wallet B deposits on chain A, if action executes, funds are transferred to wallet A
+*/
+```
+:::info
+On the chains section, we can specify any of the chains [supported](../resources/supported-chains) by LIT.
+:::
+
+You can find a generator which takes the above params to generate a Lit Action for swap: [swapActionGenerator.js](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapActionGenerator.js)
+
+Remember, once a Mint/Grant/Burn PKP is created, anyone can execute the Lit Action on the Lit Network. This execution needs to define the parameters needed by the Lit Action to run. While generating a sign or checking conditions on the action, we need information around the pkp for which we are doing it. Due to this, we need to construct our Lit Action in a way that works with any PKP.
+
+Our Lit Action primarily focuses on three factors:
+
+- Access Conditions for checking funds on both chains
+- Transaction objects for transferring funds between swap parties
+- Clawback Transaction objects for revert transferring funds to their owners
+- Lit Action conditions which decide which transaction object to sign and return
+
+We'll use [Lit's Access Control](../sdk/access-control/evm/basic-examples#must-posess-at-least-one-erc20-token) here to verify if the conditions are being met for the swap. There need to be 2 conditions for a swap to execute,
```

Comment:
"We'll use an Access Control Condition (ACC) here to verify if the swap conditions are met before it is executed. These conditions are:"